common/functions/src/scalars/strings/locate.rs

```diff
@@ -62,8 +62,18 @@ impl Function for LocateFunction {
     }
 
     fn eval(&self, columns: &DataColumnsWithField, input_rows: usize) -> Result<DataColumn> {
-        let ss_column = columns[0].column().cast_with_type(&DataType::String)?;
-        let s_column = columns[1].column().cast_with_type(&DataType::String)?;
+        let (ss_column, s_column) = if *self.display_name.to_uppercase() == "INSTR".to_owned() {
```

Comment:
A generic constant value will be better.

Reply:
@sundy-li can you explain more?

Reply:
you are right. fixed.